geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaInputFormat.scala

```diff
@@ -155,13 +157,16 @@ class GeoMesaInputFormat extends InputFormat[Text, SimpleFeature] {
    * number of shards in the schema as a proxy for number of tservers. We also know that each range will
    * only have a single location, because we always set autoAdjustRanges in the configure method above.
    */
-  override def getSplits(context: JobContext): java.util.List[InputSplit] = {
+  override def getSplits(context: JobContext) : java.util.List[InputSplit] = {
     init(context.getConfiguration)
     val accumuloSplits = delegate.getSplits(context)
-    // try to create 2 mappers per node - account for case where there are less splits than shards
-    val groupSize = Math.max(numShards * 2, accumuloSplits.length / (numShards * 2))
+    var groupSize = Math.max(numShards * 2, accumuloSplits.length / (numShards * 2)) // preserve original behavior.
+    if (desiredSplitCount != -1)
+    {
+      groupSize = Math.max(1, accumuloSplits.length / desiredSplitCount)
+    }
 
-    // We know each range will only have a single location because of autoAdjustRanges
+   // We know each range will only have a single location because of autoAdjustRanges
```

Comment:
Auto-adjust ranges merges ranges down, doesn't it? This can lead to very 
large partitions. I'll try it with it on later tonight, if you want.

On 05/19/2015 04:46 PM, Emilio wrote:
>
> In 
> geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaInputFormat.scala 
> <https://github.com/locationtech/geomesa/pull/584#discussion_r30644680>:
>
> >
> > -    // We know each range will only have a single location because of autoAdjustRanges
> > +   // We know each range will only have a single location because of autoAdjustRanges
>
> hmm, why would we not want to autoadjust ranges? if we do run it, then 
> ideally each range will be run on the appropriate local tserver. 
> Otherwise we will get more network traffic/slowness.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/locationtech/geomesa/pull/584/files#r30644680>.
>